test/metabase/api/collection_test.clj

```diff
@@ -878,9 +890,56 @@
                  :slug     "my_beautiful_collection"
                  :color    "#ABCDEF"
                  :location "/"
+                 :type     "official"
                  :parent_id nil})
                (mt/user-http-request :crowberto :put 200 (str "collection/" (u/the-id collection))
-                                     {:name "My Beautiful Collection", :color "#ABCDEF"})))))
+                                     {:name "My Beautiful Collection", :color "#ABCDEF", :type "official"})))))
+    (testing "Admins can edit the type"
+      (mt/with-temp Collection [collection]
+        (is (= "official"
+               (-> (mt/user-http-request :crowberto :put 200 (str "collection/" (u/the-id collection))
+                                         {:name "foo" :type "official"})
+                   :type))))
+      (testing "But not for personal collections"
```

Comment:
I might also be worth checking the `:type` of the object in the app DB... it's totally possible to write an endpoint that returns a version of the Collection with updated `:type` that doesn't actually persist it to the DB, and this test would still pass. Better to be safe and check both